modules/saml/src/Controller/WebBrowserSingleSignOn.php

```diff
@@ -107,6 +110,23 @@ public function artifactResolutionService(): RunnableResponse
         return new RunnableResponse([$binding, 'send'], [$artifactResponse]);
     }
 
+    public function headRequestNotAllowed(): RunnableResponse
+    {
+        Logger::info('Handling a HEAD request by returning method not allowed...');
+        
+        $request = Request::createFromGlobals();
+
+        // These are the allowed methods from routes.yml
+        $allowedMethods = ['GET', 'POST'];
+        $message = sprintf(
+            'No route found for "%s %s": Method Not Allowed (Allow: %s)',
```

Comment:
Since we want to use this controller generically for all SAML-endpoints, we cannot know the 'allowed methods' and have to make this error message a little more generic

Reply:
If there are only a handful of GET,POST combinations we could pass in the allowedMethods to the method from the routes yml so it can pass that info on. https://github.com/simplesamlphp/simplesamlphp/pull/2481/commits/b423796fd418382d97df4abc314c0182a57bf76f

Reply:
For another PR:  we should probably ditch the routes.yml files and use PHP attribute-annotations instead